server/user_account/serializers.py

```diff
@@ -1,83 +1,33 @@
 from rest_framework import serializers
-from .models import CustomUser
+from django.core.exceptions import ObjectDoesNotExist
+from .models import CustomUser, Organization
 
 
-class UserSerializer(serializers.ModelSerializer):
+class CustomUserSerializer(serializers.ModelSerializer):
     class Meta:
         model = CustomUser
-        fields = ['id', 'user_name', 'first_name', 'last_name',
-                  'password', 'role', 'email', 'location', 'is_active', 'organization']
-
-    def save(self, **kwargs):
-        """
-        Overriding the serializer save function in order to
-        access the parameters passed in the request
-        before saving them in the database.
-        In this particular case I'm hashing the password.
-        """
-
-        user = CustomUser(
-            user_name=self.validated_data['user_name'],
-            first_name=self.validated_data['first_name'],
-            last_name=self.validated_data['last_name'],
-            role=self.validated_data['role'],
-            email=self.validated_data['email'],
-            location=self.validated_data['location'],
-            is_active=self.validated_data['is_active'],
-            organization=self.validated_data['organization']
-        )
-
-        # hash password
-        password = self.validated_data['password']
-        user.set_password(password)
-
-        user.save()
-        return user
-
-
-class LoginSerializer(serializers.ModelSerializer):
-    class Meta:
-        model = CustomUser
-        fields = ['email', 'password']
-
-class LoginMobileSerializer(serializers.ModelSerializer):
-    class Meta:
-        model = CustomUser
-        fields = ['email', 'password']
-
-class ClientGridSerializer(serializers.ModelSerializer):
-    class Meta:
-        model = CustomUser
-        fields = ['user_name', 'first_name', 'last_name', 'email',
-                  'role', 'location', 'is_active', 'id']
-
-
-class UserProfileSerializer(serializers.ModelSerializer):
-    class Meta:
-        model = CustomUser
-        fields = ['user_name', 'first_name', 'last_name', 'email']
-
-
-class UserPasswordSerializer(serializers.ModelSerializer):
-    class Meta:
-        model = CustomUser
-        fields = ['password']
-
-    def save(self, **kwargs):
-        """
-        Overriding the serializer save function in order to
-        access the parameters passed in the request
-        before saving them in the database.
-        In this particular case I'm hashing the password.
-        """
-
-        # Retrieving the ID of the user that we are trying to update from the view
-        target_user_id = self.context['view'].kwargs['pk']
-        user = CustomUser.objects.get(id=target_user_id)
-
-        # hash password
-        password = self.validated_data['password']
-        user.set_password(password)
-
-        user.save()
+        fields = []
+
+    def save(self):
```

Comment:
> Can you provide in a comment what exactly are the extra checks doing here after overriding the save() method ?
> If its fields level validation django provides documentation for it:
> https://www.django-rest-framework.org/api-guide/serializers/#field-level-validation

In terms of field validation, the goal here wasn't to touch what was already implemented in the User serializer though it could be a good idea to go back through this and verify though I figured this could be left over for another PR / refactoring. As it can be seen from the original User Serializer as well as the User Password serializer, the only feature we needed is the hashing of the password. Otherwise the code first looks through the validators (self.validated_data) to see if it "can save". After this, the other two conditions are related to our model and you would need to understand it in order to comprehend what's going on. The Organization related to it is a foreign key and thus the id can't simply be plugged in as the other values. The user_name on the other hand is a unique field and I believe required thus if it is handed the assumption is a new user is being made. If not the code will fail it as this shouldn't be possible.

The reason there isn't a comment is because it is only ~20 lines of code and the assumption is that if you are working with the model, the dev would be already know about the model's fields. This could be expanded though because hard coding the user_name and Organization isn't flexible and would lead to more checks when this should all be done in an array